tests/testdata/codegen/deferred-outputs-pp/nodejs/deferred-outputs.ts

```diff
@@ -0,0 +1,15 @@
+import * as pulumi from "@pulumi/pulumi";
+import { First } from "./first";
+import { Second } from "./second";
+
+const [secondPasswordLength, resolveSecondPasswordLength] = pulumi.deferredOutput<number>();
+const first = new First("first", {passwordLength: secondPasswordLength});
+const second = new Second("second", {petName: first.petName});
+resolveSecondPasswordLength(second.passwordLength);
+const [loopingOverMany, resolveLoopingOverMany] = pulumi.deferredOutput<Array<number>>();
+const another = new First("another", {passwordLength: loopingOverMany.apply(loopingOverMany => loopingOverMany.length)});
+const many: Second[] = [];
+for (const range = {value: 0}; range.value < 10; range.value++) {
+    many.push(new Second(`many-${range.value}`, {petName: another.petName}));
+}
+resolveLoopingOverMany(pulumi.output(many.map((v, k) => [k, v]).map(([_, v]) => (v.passwordLength))));
```

Comment:
Out of curiosity, why do we generate this double map here?

Reply:
The first `.map` normalizes arrays and objects into an array and the second is the transformation. The first one can have `.filter(...)` if there is a condition variable which is performed before the transformation